src/test/java/com/google/starfish/services/FavoriteNoteServiceTest.java

```diff
@@ -0,0 +1,423 @@
+package com.google.starfish.services;
+
+import java.util.Date;
+import java.util.Calendar;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+import java.sql.SQLException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import javax.sql.DataSource;
+import static com.ninja_squad.dbsetup.Operations.*;
+import static org.junit.Assert.assertTrue;
+
+import com.google.starfish.services.FavoriteNoteService.Recency;
+import com.google.starfish.models.Note;
+import com.ninja_squad.dbsetup.DbSetup;
+import com.ninja_squad.dbsetup.destination.DataSourceDestination;
+import com.ninja_squad.dbsetup.operation.Operation;
+
+@RunWith(JUnit4.class)
+public class FavoriteNoteServiceTest {
+  private Constants constants = new Constants();
+  private String NOTES = constants.NOTES;
+  private String LABELS = constants.LABELS;
+  private String FAVORITE_NOTES = constants.FAVORITE_NOTES;
+  private String USERS = constants.USERS;
+  private DataSource pool = Constants.pool;
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+
+  // Comparator to sort Note arrays by id
+  private final Comparator<Note> COMPARE_NOTES_BY_ID = new Comparator<Note>() {
+    @Override
+    public int compare(Note n1, Note n2) {
+      return Long.compare(n1.getId(), n2.getId());
+    }
+  };
+
+  // Variables to hold the indicated dates
+  private Date TODAY = getToday();
+  private Date THIS_WEEK = getThisWeek();
+  private Date THIS_MONTH = getThisMonth();
+  private Date ALL_TIME = getAllTime();
+
+  // These variables represent the total number of rows that should be inserted 
+  // into the FAVORITE_NOTES table where the data is as indicated.
+  private final int NUM_TRENDING_NOTES_TODAY = 2;
+  private final int NUM_TRENDING_NOTES_THIS_WEEK = NUM_TRENDING_NOTES_TODAY + 2;
+  private final int NUM_TRENDING_NOTES_THIS_MONTH = NUM_TRENDING_NOTES_THIS_WEEK + 2;
```

Comment:
It doesn't look like you use these anymore?

Reply:
Yep you're right, good catch! I'll get rid of them